### PR TITLE
Clean up python dependencies

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

## What changes are proposed in this pull request?

- Remove duplicated depencies
- Separate some packages into new requirement files to optimize time to install them.

## How is this patch tested?

Verify that all the tests pass with new action config and requirements files.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for
Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by smurching — COMMENTED** [review]
Could we update test-requirements.txt to install the new split-out requirements files, e.g. `travis/flavors-requirements.txt`, so that users who follow the contributor guide & install dependencies as described [here](https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#prerequisites) will still find all the necessary dependencies are installed? LGTM otherwise

File: dev-requirements.txt [hunk 1]
```diff
@@ -1,13 +0,0 @@
-# Dev/Deployment
-sphinx==2.2.1
-sphinx_rtd_theme==0.4.3
-sphinx-autobuild==0.7.1
-sphinx-click==2.3.1
```
Comment (harupy) [inline dev-requirements.txt:5]: Moved these packages in `docs-requirements.txt`

File: dev-requirements.txt [hunk 2]
```diff
@@ -1,13 +0,0 @@
...
-gitpython>=2.1.0
-nose
-codecov
-coverage
-pypi-publisher
-scikit-learn
-scipy
-kubernetes
```
Comment (harupy) [inline dev-requirements.txt:13]: These packages are included in either `install_requires` or `small-requirements.txt`.

File: .github/workflows/master.yml [hunk 3]
```diff
@@ -16,11 +16,14 @@ jobs:
     - uses: actions/checkout@v2
     - name: Install dependencies
       env:
-        INSTALL_LARGE_PYTHON_DEPS: true
         INSTALL_SMALL_PYTHON_DEPS: true
+        INSTALL_LARGE_PYTHON_DEPS: true
+        INSTALL_FLAVORS_DEPS: true
+        INSTALL_SAGEMAKER_DEPS: true
+        INSTALL_TF_DEPS: true
+        INSTALL_LINT_DEPS: true
```
Comment (harupy) [inline .github/workflows/master.yml:24]: This looks a bit ugly but inevitable.

Comment (harupy) [inline .github/workflows/master.yml:24]: This looks a bit ubly but ienvitable

File: travis/small-requirements.txt [hunk 4]
```diff
@@ -1,22 +1,16 @@
...
-azure-storage-blob==12.3.0
 google-cloud-storage==1.14.0
-botocore==1.12.84  # pinned for moto
-boto3==1.9.84      # pinned for moto
-mock==2.0.0
+botocore==1.12.84
+boto3==1.9.84
 moto==1.3.7
-mxnet==1.5.0
 pandas<=0.23.4
```
Comment (harupy) [inline travis/small-requirements.txt:6]: Couldn't remove pandas here due to this issue https://github.com/mlflow/mlflow/issues/3006.

Comment (harupy) [inline travis/small-requirements.txt:6]: pandas 23.4 is very old (released on Aug 5, 2018). Maybe we should update it to a newer version.

Comment (harupy) [reply travis/small-requirements.txt:6]: #3130 removed this restriction

**Pull request closed without merging**